src/main/java/net/javadiscord/javabot/systems/qotw/QOTWPointsService.java

```diff
@@ -121,15 +124,34 @@ public List<QOTWAccount> getTopAccounts(int amount, int page) {
 	/**
 	 * Increments a single user's QOTW-Points.
 	 *
-	 * @param userId The discord Id of the user.
+	 * @param member The {@link Member} whose points shall be incremented
 	 * @return The total points after the update.
 	 */
-	public long increment(long userId) {
+	public long increment(Member member) {
+		long userId = member.getIdLong();
 		try {
 			LocalDate date=LocalDate.now();
 			int points = pointsRepository.getPointsAtDate(userId, date)+1;
 			pointsRepository.setPointsAtDate(userId, date, points);
-			return pointsRepository.getByUserId(userId, getCurrentMonth()).map(QOTWAccount::getPoints).orElse(0L);
+			Role qotwChampionRole = botConfig.get(member.getGuild()).getQotwConfig().getQOTWChampionRole();
+			LocalDate month = getCurrentMonth();
+			long newScore = pointsRepository.getByUserId(userId, month).map(QOTWAccount::getPoints).orElse(0L);
+			if (qotwChampionRole != null) {
+				pointsRepository.getTopAccounts(month, 0, 1)
+					.stream()
+					.findFirst()
+					.ifPresent(best -> {
+						if (newScore >= best.getPoints()) {
+							member.getGuild().addRoleToMember(member, qotwChampionRole).queue();
+							for (Member m : member.getGuild().getMembersWithRoles(qotwChampionRole)) {
+								if (getOrCreateAccount(m.getIdLong()).getPoints() < best.getPoints()) {
+									m.getGuild().removeRoleFromMember(m, qotwChampionRole).queue();
+								}
+							}
+						}
+					});
+			}
+			return newScore;
```

Comment:
What about scheduling a job that does this at the end of each month? It's the "QOTW Champion" of the last month then

Reply:
Done